utility/src/test/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImplTest.kt

```diff
@@ -208,9 +226,138 @@ class PerformanceMetricsAssessorImplTest {
+      "secondAppTimeValue=186933790700",
+      "secondNumberOfOnlineCoresValue=3",
+      "relativeCpuUsage=0.00000000239"
+    ),
+  )
+  fun testAssessor_setFirstAndSecondSnapshot_verifyCorrectCpuUsageIsReturned() {
+    val firstSnapshot = PerformanceMetricsAssessor.CpuSnapshot(
+      TEST_FIRST_PROCESS_TIME,
+      TEST_FIRST_CPU_TIME,
+      TEST_FIRST_NUMBER_OF_CORES
+    )
+    val secondSnapshot = PerformanceMetricsAssessor.CpuSnapshot(
+      secondAppTimeValue,
+      secondCpuValue,
+      secondNumberOfOnlineCoresValue
+    )
+    val returnedRelativeCpuUsage =
+      performanceMetricsAssessorImpl.getRelativeCpuUsage(firstSnapshot, secondSnapshot)
+    assertThat(returnedRelativeCpuUsage).isWithin(1e-5).of(relativeCpuUsage)
+  }
+
+  @Test
+  @RunParameterized(
+    Iteration(
+      "negativeDeltaCpuValue",
+      "secondCpuValue=900",
+      "secondAppTimeValue=1665790700",
+      "secondNumberOfOnlineCoresValue=2"
+    ),
+    Iteration(
+      "negativeCpuValue",
+      "secondCpuValue=-900",
+      "secondAppTimeValue=1665790700",
+      "secondNumberOfOnlineCoresValue=2"
+    ),
+    Iteration(
+      "negativeDeltaAppTimeValue",
+      "secondCpuValue=1200",
+      "secondAppTimeValue=1665790050",
+      "secondNumberOfOnlineCoresValue=2"
+    ),
+    Iteration(
+      "negativeAppTimeValue",
+      "secondCpuValue=1200",
+      "secondAppTimeValue=-1665790050",
+      "secondNumberOfOnlineCoresValue=2"
+    ),
+    Iteration(
+      "zeroDeltaAppTimeValue",
+      "secondCpuValue=1200",
+      "secondAppTimeValue=1665790650",
+      "secondNumberOfOnlineCoresValue=2"
+    ),
+    Iteration(
+      "zeroDeltaOnlineCores",
+      "secondCpuValue=1200",
+      "secondAppTimeValue=1665790700",
+      "secondNumberOfOnlineCoresValue=0"
+    ),
+    Iteration(
+      "negativeOnlineCores",
+      "secondCpuValue=1200",
+      "secondAppTimeValue=1665790700",
+      "secondNumberOfOnlineCoresValue=-1"
+    ),
+    Iteration(
+      "outOfBoundsCpuUsageValue",
+      "secondCpuValue=9223372036854775807",
+      "secondAppTimeValue=1665790700",
+      "secondNumberOfOnlineCoresValue=2"
+    ),
+  )
+  fun testAssessor_inputFailureSystemValues_calculateCpuUsage_verifyFailure() {
+    val firstSnapshot = PerformanceMetricsAssessor.CpuSnapshot(
+      TEST_FIRST_PROCESS_TIME,
+      TEST_FIRST_CPU_TIME,
+      TEST_FIRST_NUMBER_OF_CORES
+    )
+
+    val secondSnapshot = PerformanceMetricsAssessor.CpuSnapshot(
+      secondAppTimeValue,
+      secondCpuValue,
+      secondNumberOfOnlineCoresValue
+    )
+
+    val relativeCpuUsage =
+      performanceMetricsAssessorImpl.getRelativeCpuUsage(firstSnapshot, secondSnapshot)
+    val exception = fakeExceptionLogger.getMostRecentException()
+
+    assertThat(relativeCpuUsage).isWithin(1e-5).of(-1.00)
+    assertThat(exception).isInstanceOf(IllegalArgumentException::class.java)
+  }
+
+  @Test
+  fun testAssessor_getCurrentCpuSnapshot_verifySnapshotReturnsCurrentValues() {
```

Comment:
This seems to be verifying time specifically, so perhaps ``testAssessor_getCurrentCpuSnapshot_snapshotHasCurrentAppTime`` would be clearer?

Please update the other test names to be more clear on the specific behaviors being tested (and, if you can't, then it may mean more than one behavior is being tested). Also, 'verify' is redundant for the 'expected outcome' portion of the test name.

Reply:
done.